app/controllers/oauth/oauth_controller.rb

```diff
@@ -5,11 +5,19 @@ class OauthController < ApplicationController
     def authorize
       validate_authorization_request(params)
 
-      code = oauth_client(params[:client_id]).create_authorization_code!(params[:redirect_uri])
+      current_user = User.find(params[:user_id])
```

Comment:
I think this business logic (that only exists because we're skipping the actual auth) would be better in the `create_authorization_code` logic of the client, that way we keep our controller slim and keep code that would ultimately get remove in the case of a full implementation out of it. You'd still want to instantiate the `current_user` object like you're doing here to handle the validations here & pass it to the `create_authorization_code` method like you are.